File: java/StudentRegistration.java
import java.util.Scanner;

public class StudentRegistration {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Prompt for student name
        System.out.print("Enter student name: ");
        String name = scanner.nextLine();

        // Prompt for number of registered units
        System.out.print("Enter number of registered units: ");
        int units = scanner.nextInt();

        // Determine registration status
        String status;
        if (units > 7) {
            status = "Overload - Approval Required";
        } else {
            status = "Registration Accepted";
        }

        // Print final summary
        System.out.printf("\nStudent Name: %s\nRegistered Units: %d\nStatus: %s\n", name, units, status);

        // Close scanner
        scanner.close();
    }
}

File: c/student_registration.c
#include <stdio.h>

int main() {
    char name[100];
    int units;
    
    // Prompt for student name
    printf("Enter your name: ");
    fgets(name, sizeof(name), stdin);
    
    // Prompt for number of registered units
    printf("Enter number of registered units: ");
    scanf("%d", &units);
    
    // Check registration status
    if (units > 7) {
        printf("Overload - Approval Required\n");
    } else {
        printf("Registration Accepted\n");
    }
    
    // Print final summary
    printf("\nFinal Summary:\n");
    printf("Student Name: %s", name);
    printf("Registered Units: %d\n", units);
    printf("Status: %s\n", (units > 7) ? "Overload - Approval Required" : "Registration Accepted");
    
    return 0;
}